CreateReport/index.ts

```diff
@@ -37,16 +39,59 @@ const reportsApi: AzureFunction = async (_context: Context, req: HttpRequest, us
             throw new ValidationFunctionError(functionName, validationError.message);
         }
 
-        const { startDate, endDate } = req.body;
+        const { startDate, endDate, unitIds, nodeGroupIds } = req.body;
 
-        const events: Event[] = await EventModel.find({ time: { $gte: startDate, $lte: endDate } }).exec();
+        const events: Event[] = await EventModel.find({ time: { $gte: startDate, $lte: endDate } })
+            .populate('nodeData')
+            .exec();
 
-        if (!events.length) {
-            throw new ObjectNotFoundFunctionError(functionName, 'events');
+        let filteredEvents: Array<{}> = [];
+
+        switch (user.role) {
+            case Role.Mada:
+            case Role.Itur:
+            case Role.Technical:
+                if (unitIds) {
+                    filteredEvents = events.filter((event: any) => unitIds.includes(event.nodeData.unitId));
+                } else {
+                    filteredEvents = events;
+                }
+                break;
+
+            case Role.RamadIturAssistant:
+            case Role.ProfessionalRamad:
+                if (!nodeGroupIds || nodeGroupIds.length < 1) {
+                    throw new ValidationFunctionError(functionName, 'nodeGroupIds is missing');
+                }
+                nodeGroupIds.forEach(async (nodeGroupId) => {
+                    const nodeGroup: NodeGroup = await nodeGroupModel.findById(nodeGroupId).exec();
+                    if (!nodeGroup || !nodeGroup.usersIds.includes(user.id) || nodeGroup.unitId !== user.unitId) {
+                        throw new ValidationFunctionError(functionName, 'User is not unauthorized');
+                    }
+                    filteredEvents = events.filter((event: any) => nodeGroupIds.includes(event.nodeData.nodeGroupId));
+                });
+                break;
+
+            case Role.RamadIturOfUnit:
+                if (nodeGroupIds) {

```

Comment:
```suggestion
                if (nodeGroupIds?.length) {
```
because if it is an empty array there wouldn't be any events